File: src/main/java/com/github/xumeng/java/problems/easy/problem0228/Solution0228.java
package com.github.xumeng.java.problems.easy.problem0228;

import java.util.ArrayList;
import java.util.List;

public class Solution0228 {

    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<>();
        if (nums.length == 0) {
            return result;
        }
        int start = 0;
        int end = start;
        while (end + 1 < nums.length) {
            if (nums[end] + 1 == nums[end + 1]) {
                end++;
            } else {
                String element = start == end ? String.valueOf(nums[start]) : nums[start] + "->" + nums[end];
                result.add(element);
                start = end + 1;
                end = start;
            }
        }
        String element = start == end ? String.valueOf(nums[start]) : nums[start] + "->" + nums[end];
        result.add(element);
        return result;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0228/Solution0228.kt
package com.github.xumeng.kotlin.problems.easy.problem0228

class Solution0228 {

    fun summaryRanges(nums: IntArray): List<String> {
        val result: MutableList<String> = mutableListOf()
        if (nums.isEmpty()) {
            return result
        }
        var start = 0
        var end = start
        while (end + 1 < nums.size) {
            if (nums[end] + 1 == nums[end + 1]) {
                end++
            } else {
                val element: String = if (start == end) {
                    nums[start].toString()
                } else {
                    nums[start].toString() + "->" + nums[end].toString()
                }
                result.add(element)
                start = end + 1
                end = start
            }
        }
        val element: String = if (start == end) {
            nums[start].toString()
        } else {
            nums[start].toString() + "->" + nums[end].toString()
        }
        result.add(element)
        return result
    }

}